### ISSUE TITLE
Feature CachingHeaders: Overload resolution ambiguity

### ISSUE THREAD

**Issue by erdlet** [ktorio/ktor#741]
### JVM Version, Operating System and Relevant Context
**JVM Version:**
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)

**Operating System:** Fedora 29

gradle.properties
```
ktor_version=1.0.0
kotlin.code.style=official
kotlin_version=1.3.10
logback_version=1.2.1
```

build.gradle
```
buildscript {
    repositories {
        jcenter()
    }

    dependencies {
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}

apply plugin: 'kotlin'
apply plugin: 'application'

group 'kaccounts'
version '0.0.1'
mainClassName = "io.ktor.server.netty.EngineMain"

sourceSets {
    main.kotlin.srcDirs = main.java.srcDirs = ['src']
    test.kotlin.srcDirs = test.java.srcDirs = ['test']
    main.resources.srcDirs = ['resources']
    test.resources.srcDirs = ['testresources']
}

repositories {
    mavenLocal()
    jcenter()
    mavenCentral()
}

ext {
    junit_jupiter_version = "5.3.1"
    commons_validator_version = "1.6"
}

dependencies {
    implementation(
            "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version",
            "io.ktor:ktor-server-netty:$ktor_version",
            "ch.qos.logback:logback-classic:$logback_version",
            "io.ktor:ktor-auth:$ktor_version",
            "commons-validator:commons-validator:$commons_validator_version"
    )

    testImplementation(
            "io.ktor:ktor-server-tests:$ktor_version",
            "org.jetbrains.kotlin:kotlin-test-junit5:$kotlin_version",
            "org.junit.jupiter:junit-jupiter-api:$junit_jupiter_version"
    )

    testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:$junit_jupiter_version"
}

test {
    useJUnitPlatform()
    reports {
        junitXml.enabled = true
        html.enabled = false
    }

}

tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
    kotlinOptions {
        jvmTarget = "1.8"
    }
}
```

### Feedback
In my application I try to add CachingHeaders to enforce no caching for my HTML pages which are genereated by the servers HTTP resources. Unfortunately I'm not able to configure `CachingOptions`, as it seems to exist as function and as class. This leads to this exception during compilation:

```
Overload resolution ambiguity: 
public constructor CachingOptions(cacheControl: CacheControl? = ..., expires: GMTDate? = ...) defined in io.ktor.http.content.CachingOptions
public fun CachingOptions(cacheControl: CacheControl? = ..., expires: ZonedDateTime? = ...): CachingOptions defined in io.ktor.http.content
```
The class / function is received from `io.ktor:ktor-http-jvm` or `io.ktor:ktor-http`. 

Is there any solution based on my dependencies (which may be wrong configured)?

**Comment by cy6erGn0m** [comment]
Due to this issue it is impossible to not specify `expires`. Workaround is to always specify `expires`. Not sure if it is always possible

**Comment by cy6erGn0m** [comment]
My suggestion is to remove function's default argument and nullability but keep in the constructor. What do you think, @e5l ?

**Comment by erdlet** [comment]
I had a look on the function in the ktor codebase and saw only one usage of it within `io.ktor.tests.server.features.CompressionTest`. Is there any other use case I'm not aware of?

If not, wouldn't it make sense to (re)move / rename the function as it doesn't do anything else than converting ZonedDateTime to GMTDate before construction of CachingOptions?

**Comment by e5l** [comment]
Probably we should deprecate one and provide clear factory methods. 
Probably we should discuss signatures.

**Comment by erdlet** [comment]
IMHO this sounds like the cleanest and most understandable solution.

**Comment by cy6erGn0m** [comment]
Fixed in 1.0.1
